### PR TITLE
Homework 17

### PR DESCRIPTION
**Pull request by aleksacastle**
Done

### CODE REVIEW

**Review by oleksandr-kychun — COMMENTED** [review]
nice try) one thing to think about - there are some 'magic numbers'

**Comment by aleksacastle** [comment]
didn't know that creating another methods was acceptable. working on the
optimisation right now

2017-06-08 11:21 GMT+03:00 ghostinshadow <notifications@github.com>:

**Comment by aleksacastle** [comment]
@ghostinshadow how about now? Was trying to implement iterator .times, but constantly failed. Don't know even why

**Review by oleksandr-kychun — COMMENTED** [review]
I like it) it reads nicely! what about numbers 65 and 26?) i can guess what is that but somebody wont)

**Review by oleksandr-kychun — COMMENTED** [review]
that's a lot better now) just move assigning of constants out of method rand_char and you're good to go

**Comment by aleksacastle** [comment]
@ghostinshadow thanks!

File: homework.rb [hunk 1]
```diff
@@ -1,6 +1,21 @@
-VARIANT = 0
+VARIANT = 17
+  A_ENCODING = 65
+  TOTAL_LETTERS_ALPHABET = 26
+  def rand_int
+    #generate random number
+    rand(10).to_s
```
Comment (sveredyuk) [inline homework.rb]: magic number

File: homework.rb [hunk 2]
```diff
@@ -1,6 +1,21 @@
-VARIANT = 0
+VARIANT = 17
+  A_ENCODING = 65
```
Comment (sveredyuk) [inline homework.rb]: no need 2 spaces and also below

File: homework.rb [hunk 3]
```diff
@@ -1,6 +1,21 @@
...
+    #generating random letter
+    (A_ENCODING + rand(TOTAL_LETTERS_ALPHABET)).chr.to_s
+  end
+
+  def generate
   # put your solution here (delete the line below)
-  rand(1000000).to_s + '-' + rand(1000000).to_s
-end
+  #template "1XC1-A11D-1CDVB-123"
+  rand_int + rand_char + rand_char + rand_int + "-" + #1XC1-
```
Comment (sveredyuk) [inline homework.rb]: wow. wow... please, try to think how you could cut all this funny algebra

**Pull request closed without merging**